src/Microsoft.ML.Core/Data/DateTime.cs

```diff
@@ -32,22 +32,19 @@ public DvDateTime(SysDateTime sdt)
         /// <summary>
         /// This ctor accepts any value for ticks, but produces an NA if ticks is out of the legal range.
         /// </summary>
-        public DvDateTime(DvInt8 ticks)
+        public DvDateTime(long ticks)
         {
-            if ((ulong)ticks.RawValue > MaxTicks)
-                _ticks = DvInt8.NA;
-            else
-                _ticks = ticks;
+            _ticks = ticks;
             AssertValid();
         }
 
         [Conditional("DEBUG")]
         internal void AssertValid()
         {
-            Contracts.Assert((ulong)_ticks.RawValue <= MaxTicks || _ticks.IsNA);
+            Contracts.Assert((ulong)_ticks <= MaxTicks);
```

Comment:
>Contracts.Assert((ulong)_ticks <= MaxTicks); [](start = 12, length = 44)

If you no longer have the bounds checking in the constructor, you cannot assert this here.

Reply:
I didn't care too much about this piece of code because DvDateTime PR will be deleting this whole file.

---
In reply to: [212032344](https://github.com/dotnet/machinelearning/pull/683#discussion_r212032344) [](ancestors = 212032344)